include/effects/matrix/PatternWeather.h

```diff
@@ -350,45 +408,81 @@ class PatternWeather : public LEDStripEffect
             vTaskDelay(pdMS_TO_TICKS(WEATHER_CHECK_WIFI_WAIT));
         }
 
-        updateCoordinates();
+        // Only try to update if we have an API Key
+        if (!g_ptrSystem->DeviceConfig().GetOpenWeatherAPIKey().isEmpty())
+        {
+            updateCoordinates();
 
-        if (getWeatherData())
-            getTomorrowTemps(highTomorrow, loTomorrow);
+            if (getWeatherData())
+                getTomorrowTemps(highTomorrow, loTomorrow);
+        }
     }
 
+    /**
+     * @brief Determine if the device configuration has changed the
+     * location selection
+     * 
+     * @return bool - true if the location has changed 
+     */
     bool HasLocationChanged()
     {
-        String configLocation = g_ptrSystem->DeviceConfig().GetLocation();
-        String configCountryCode =g_ptrSystem->DeviceConfig().GetCountryCode();
+        bool locationChanged = g_ptrSystem->DeviceConfig().GetLocation() != strLocation;
+        bool contryChanged = g_ptrSystem->DeviceConfig().GetCountryCode() != strCountryCode;
 
-        return strLocation != configLocation || strCountryCode != configCountryCode;
+        return locationChanged || contryChanged;
     }
 
 public:
 
+    /**
+     * @brief Construct a new Pattern Weather object
+     * 
+     */
     PatternWeather() : LEDStripEffect(EFFECT_MATRIX_WEATHER, "Weather")
     {
     }
 
+    /**
+     * @brief Construct a new Pattern Weather object
+     * 
+     * @param jsonObject Configuration JSON Object
+     */
     PatternWeather(const JsonObjectConst&  jsonObject) : LEDStripEffect(jsonObject)
     {
     }
 
+    /**
+     * @brief Destroy the Pattern Weather object
+     * Cancel the Network Reader thread for this index
+     */
     ~PatternWeather()
     {
         g_ptrSystem->NetworkReader().CancelReader(readerIndex);
     }
 
+    /**
+     * @brief 
+     * 
+     * @param gfx Graphic Base engine
+     * @return bool - true if successfully initialized

```

Comment:
In earlier documentation blocks you didn't mention the bool return type if you mentioned one of the bool values ("false" in the case of `ShouldShowTitle()`, `RequiresDoubleBuffering()`, etc.) I think that makes sense. You could opt to include the bool return type here, but then I think it's best to do it everywhere.

Reply:
Yeah, I went with putting the type in as well as a description everywhere.